src/alchemlyb/tests/test_units.py

```diff
@@ -0,0 +1,184 @@
+        dataset = load_benzene()
+        u_nk = extract_u_nk(dataset['data']['Coulomb'][0], 310)
+        new_u_nk = slicing(u_nk)
+        assert new_u_nk.attrs['temperature'] == 310
+        assert new_u_nk.attrs['energy_unit'] == 'kT'
+
+    def test_statistical_inefficiency(self, dhdl):
+        '''Test if extract_u_nk assign the attr correctly'''
+        dataset = load_benzene()
+        dhdl = extract_dHdl(dataset['data']['Coulomb'][0], 310)
+        new_dhdl = statistical_inefficiency(dhdl)
+        assert new_dhdl.attrs['temperature'] == 310
+        assert new_dhdl.attrs['energy_unit'] == 'kT'
+
+    def test_equilibrium_detection(self, dhdl):
+        '''Test if extract_u_nk assign the attr correctly'''
+        dataset = load_benzene()
+        dhdl = extract_dHdl(dataset['data']['Coulomb'][0], 310)
+        new_dhdl = equilibrium_detection(dhdl)
+        assert new_dhdl.attrs['temperature'] == 310
+        assert new_dhdl.attrs['energy_unit'] == 'kT'
+
+class Test_estimator():
+    '''Test the estimator module.'''
+    @staticmethod
+    @pytest.fixture(scope='class')
+    def dhdl():
+        bz = load_benzene().data
+        dHdl_coul = pd.concat(
+            [extract_dHdl(xvg, T=300) for xvg in bz['Coulomb']])
+        dHdl_coul.attrs = extract_dHdl(load_benzene().data['Coulomb'][0], T=300).attrs
+        return dHdl_coul
+
+    @staticmethod
+    @pytest.fixture(scope='class')
+    def u_nk():
+        bz = load_benzene().data
+        u_nk_coul = pd.concat(
+            [extract_u_nk(xvg, T=300) for xvg in bz['Coulomb']])
+        u_nk_coul.attrs = extract_dHdl(load_benzene().data['Coulomb'][0], T=300).attrs
+        return u_nk_coul
+
+    def test_ti(self, dhdl):
+        ti = TI().fit(dhdl)
+        assert ti.delta_f_.attrs['temperature'] == 300
+        assert ti.delta_f_.attrs['energy_unit'] == 'kT'
+        assert ti.d_delta_f_.attrs['temperature'] == 300
+        assert ti.d_delta_f_.attrs['energy_unit'] == 'kT'
+        assert ti.dhdl.attrs['temperature'] == 300
+        assert ti.dhdl.attrs['energy_unit'] == 'kT'
+
+    def test_ti_separate_dhdl(self, dhdl):
+        ti = TI().fit(dhdl)
+        dhdl_list = ti.separate_dhdl()
+        for dhdl in dhdl_list:
+            assert dhdl.attrs['temperature'] == 300
+            assert dhdl.attrs['energy_unit'] == 'kT'
+
+    def test_bar(self, u_nk):
+        bar = BAR().fit(u_nk)
+        assert bar.delta_f_.attrs['temperature'] == 300
+        assert bar.delta_f_.attrs['energy_unit'] == 'kT'
+        assert bar.d_delta_f_.attrs['temperature'] == 300
+        assert bar.d_delta_f_.attrs['energy_unit'] == 'kT'
+
+    def test_mbar(self, u_nk):
+        mbar = MBAR().fit(u_nk)
+        assert mbar.delta_f_.attrs['temperature'] == 300
+        assert mbar.delta_f_.attrs['energy_unit'] == 'kT'
+        assert mbar.d_delta_f_.attrs['temperature'] == 300
+        assert mbar.d_delta_f_.attrs['energy_unit'] == 'kT'
+
+class Test_visualisation():
```

Comment:
add as `TestUnits` to the `test_visualisation.py` so that in the future it is easier to add tests when anything changes in viz.